File: Server/Core/Entities/Message.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Core.Entities
{
    public class Message
    {
        [Key]
        public int Id { get; set; }
        [MaxLength(750)]
        public string Text { get; set; }
        public DateTime SentDate { get; set; }
        public Chat Chat { get; set; }
        [ForeignKey("Chat")]
        public int ChatId { get; set; }
        public Contact Sender { get; set; }
        [ForeignKey("Sender")]
        public int SenderId { get; set; }
    }
}

File: Client/chapar/lib/DTOs/message.dart
class Message {
  int id;
  String text;
  String sentDate;
  int senderId;
  int chatId;
  bool? isDeliverd = true;
  Message(this.id, this.chatId, this.text, this.senderId, this.sentDate,
      {this.isDeliverd});
  factory Message.fromJson(Map<String, dynamic> json) {
    return Message(
      json['id'],
      json['chatId'],
      json['text'],
      json['senderId'],
      json['sentDate'],
    );
  }
}
